### ISSUE TITLE
ClassCastException: clojure.lang.PersistentHashMap cannot be cast to com.intellij.openapi.util.TextRange

### ISSUE THREAD

**Issue by lorddoig** [cursiveclojure/cursive#754]
FYI: The following is about ClojureScript code.

Something changed in the `0.1.46` release that's giving me these exceptions pretty much constantly (still present `0.1.47`).  Eventually the IDE just gives up and marks all new code with that wavy red underline until it's restarted (lots of `Incorrect arity 0 for defn` errors).

I'm not sure if it's related, but after updating to these versions `Reformat Code` operations in some projects completely lock up IntelliJ for 10-30 seconds.  Doing so in `src/cljs/ns/core.cljs` of a new leiningen project based on the `reagent` template seems to exhibit this behaviour.  These projects also get my MacBook's fans going during indexing like no others.

```
clojure.lang.PersistentHashMap cannot be cast to com.intellij.openapi.util.TextRange
java.lang.ClassCastException: clojure.lang.PersistentHashMap cannot be cast to com.intellij.openapi.util.TextRange
	at cursive.psi.impl.synthetic.ClJavascriptSymbol.getTextRange(ClJavascriptSymbol.java:157)
	at plugin.psi$text_range.invoke(psi.clj:69)
	at plugin.symbol$var_meta.invoke(symbol.clj:45)
	at clojure.core$map$fn__4529.invoke(core.clj:2614)
	at clojure.lang.LazySeq.sval(LazySeq.java:40)
	at clojure.lang.LazySeq.seq(LazySeq.java:49)
	at clojure.lang.LazySeq.first(LazySeq.java:71)
	at clojure.lang.RT.first(RT.java:625)
	at clojure.core$first__4091.invoke(core.clj:55)
	at plugin.inspections.list$incorrect_arity.invoke(list.clj:75)
	at clojure.lang.Var.invoke(Var.java:388)
	at cursive.api.DelayedFn.invoke(DelayedFn.java:41)
	at cursive.inspections.ClInspectionVisitor.visitElement(ClInspectionVisitor.java:32)
	at com.intellij.psi.impl.PsiElementBase.accept(PsiElementBase.java:274)
	at com.intellij.codeInspection.InspectionEngine.acceptElements(InspectionEngine.java:80)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass$4.process(LocalInspectionsPass.java:313)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass$4.process(LocalInspectionsPass.java:308)
	at com.intellij.concurrency.ApplierCompleter.a(ApplierCompleter.java:122)
	at com.intellij.concurrency.ApplierCompleter.access$000(ApplierCompleter.java:44)
	at com.intellij.concurrency.ApplierCompleter$1.run(ApplierCompleter.java:85)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1146)
	at com.intellij.concurrency.ApplierCompleter$2.run(ApplierCompleter.java:94)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.a(ProgressManagerImpl.java:281)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.a(ProgressManagerImpl.java:278)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:233)
	at com.intellij.concurrency.ApplierCompleter.a(ApplierCompleter.java:106)
	at com.intellij.concurrency.ApplierCompleter.compute(ApplierCompleter.java:82)
	at jsr166e.CountedCompleter.exec(CountedCompleter.java:684)
	at jsr166e.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at jsr166e.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:858)
	at jsr166e.ForkJoinPool.scan(ForkJoinPool.java:1687)
	at jsr166e.ForkJoinPool.runWorker(ForkJoinPool.java:1642)
	at jsr166e.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:109)
```

**Comment by romstad** [comment]
I get this one, too. I've discovered that I can reduce the contents of core.cljs down to the following and still get the error:

```clojure
(ns foo.core
    (:import goog.History))

(defn hook-browser-navigation! []
  (doto (History.)))
```

**Comment by lorddoig** [comment]
I tried narrowing it down like this but seem to have missed `goog.History`, so it seems that's the likely culprit of the reformatting lock-ups.  The exceptions/giving up on code analysis appear to be independent of this though.

**Comment by cursiveclojure** [comment]
Sorry about this one, it's fixed for the next build - I'll get that out soon. This was introduced fixing #735. However I can't reproduce the formatting issue - once this problem is fixed, if that's still happening could you open another issue for it?

**Comment by romstad** [comment]
Just to be clear, I've only seen the "clojure.lang.PersistentHashMap cannot be cast to com.intellij.openapi.util.TextRange" bug, not the formatting issue. Glad to hear it will be fixed in the next version!  :-)

**Comment by cursiveclojure** [comment]
I've just released 0.1.48, which contains this fix. However, I've just realised that I neglected to bump the index versions which would cause them to be automatically rebuilt. If you're affected by this problem, *File->Invalidate Caches and Restart* will cause the indexes to be built from scratch and the problem should be fixed.

I really need to make this an automatic part of my release process - sorry for the inconvenience!
